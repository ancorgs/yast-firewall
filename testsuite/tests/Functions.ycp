/****************************************************************************

Copyright (c) 2000 - 2012 Novell, Inc.
All Rights Reserved.

This program is free software; you can redistribute it and/or
modify it under the terms of version 2 of the GNU General Public License as
published by the Free Software Foundation.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program; if not, contact Novell, Inc.

To contact Novell about this file by physical or electronic mail,
you may find current contact information at www.novell.com

****************************************************************************/
{
    include "testsuite.ycp";
    
    include "firewall/subdialogs.ycp";
    include "firewall/uifunctions.ycp";

    map READ = $[
	"target" : $[
	    "tmpdir" : "/tmp",
	],
    ];
    map WRITE = $[];
    map EXECUTE = $[];

    TESTSUITE_INIT([READ, WRITE, EXECUTE], nil);

    DUMP ("");

    DUMP ("== Checking Functions ==");

    TEST (``(CheckNetwork("192.168.0.1")), [READ, WRITE, EXECUTE], nil);
    TEST (``(CheckNetwork("192.168.0.255")), [READ, WRITE, EXECUTE], nil);
    TEST (``(CheckNetwork("192.168.0.1/20")), [READ, WRITE, EXECUTE], nil);
    TEST (``(CheckNetwork("192.168.0.255/32")), [READ, WRITE, EXECUTE], nil);
    TEST (``(CheckNetwork("192.168.0.1/255.240.0.0")), [READ, WRITE, EXECUTE], nil);
    TEST (``(CheckNetwork("192.168.0.255/255.255.255.255")), [READ, WRITE, EXECUTE], nil);
    TEST (``(CheckNetwork("192.168.0.1/0")), [READ, WRITE, EXECUTE], nil);
    TEST (``(CheckNetwork("172.55.0.0/1")), [READ, WRITE, EXECUTE], nil);
    TEST (``(CheckNetwork("0/0")), [READ, WRITE, EXECUTE], nil);
    TEST (``(CheckNetwork("255.255.255.255/255.255.255.255")), [READ, WRITE, EXECUTE], nil);

    TEST (``(CheckNetwork("172.55.0.0/33")), [READ, WRITE, EXECUTE], nil);
    TEST (``(CheckNetwork("256.168.0.255")), [READ, WRITE, EXECUTE], nil);
    TEST (``(CheckNetwork("172.55.0.0/125.85.5.5")), [READ, WRITE, EXECUTE], nil);
    TEST (``(CheckNetwork("192.168.0.255/5.5.5")), [READ, WRITE, EXECUTE], nil);
    TEST (``(CheckNetwork("192.168.0.255/255.255.0.255")), [READ, WRITE, EXECUTE], nil);

    DUMP ("");
}