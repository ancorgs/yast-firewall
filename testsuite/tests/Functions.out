Read	.target.tmpdir nil
Log	Failed to set temporary directory: nil
Dir	.product.features.section: []
Dump	
Dump	== Checking Functions ==
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	false
Return	false
Return	false
Return	false
Return	false
Dump	
